### PR TITLE
[Importer] adding hasHeader, file_url to improve the code

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

adding hasHeader feature
and using file_url in the place of csv_data (global variable)
adding unittest

### CODE REVIEW

File: desktop/libs/indexer/src/indexer/api3.py [hunk 1]
```diff
@@ -733,12 +738,9 @@ def upload_local_file(request):
 
   upload_file = request.FILES['inputfile']
   fs = FileSystemStorage()
-  name = fs.save(upload_file.name, upload_file)
-  reader = csv.reader(decode_utf8(upload_file))
-
-  csv_data.clear()
+  filename = "%s.%s" % (uuid.uuid4(), 'csv')
+  name = fs.save(filename, upload_file)
```
Comment (agl29) [inline desktop/libs/indexer/src/indexer/api3.py:742]: unique id is given to file name

Comment (romainr) [reply desktop/libs/indexer/src/indexer/api3.py:742]: Also username prefix?

File: desktop/libs/indexer/src/indexer/indexers/sql.py [hunk 2]
```diff
@@ -266,7 +268,7 @@ def create_table_from_a_file(self, source, destination, start_time=-1):
         is_task=True
     )
 
-  def create_table_from_local_file(self, source, destination, csv_data, start_time=-1):
+  def create_table_from_local_file(self, source, destination, start_time=-1):
```
Comment (romainr) [inline desktop/libs/indexer/src/indexer/indexers/sql.py:271]: Probably want `local_file` in input and add a test for this?

File: desktop/libs/indexer/src/indexer/templates/importer.mako [hunk 3]
```diff
@@ -3161,7 +3161,7 @@ ${ commonheader(_("Importer"), "indexer", user, request, "60px") | n,unicode }
           var files = $('#inputfile')[0].files[0];
           fd.append('inputfile', files);
           var file_size = files.size;
-          if (file_size < 30*1000) {
+          if (file_size < 300*1000) {
```
Comment (romainr) [inline desktop/libs/indexer/src/indexer/templates/importer.mako:3164]: Let's remove this for now or target 10MB

**Review by romainr — APPROVED** [review]

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py [hunk 4]
```diff
@@ -603,3 +603,52 @@ def test_create_ddl_with_abfs():
   finally:
     finish()
   assert_true(u"\'abfs://my-data@yingstorage.dfs.core.windows.net/test_data/cars.csv\'" in sql)
+
+
+def test_create_table_from_local():
```
Comment (agl29) [inline desktop/libs/indexer/src/indexer/indexers/sql_tests.py:608]: unittest added

File: desktop/libs/indexer/src/indexer/indexers/sql.py [hunk 5]
```diff
@@ -292,13 +294,20 @@ def create_table_from_local_file(self, source, destination, csv_data, start_time
...
+        list_of_tuples = list(map(tuple, reader))
+
+        if source['format']['hasHeader']:
+          list_of_tuples = list_of_tuples[1:]
+
+        csv_rows = str(list_of_tuples)[1:-1]
+        sql += '''INSERT INTO %(table_name)s VALUES %(csv_rows)s;'''% {
+            'table_name': table_name,
+            'csv_rows': csv_rows
+          }
```
Comment (agl29) [inline desktop/libs/indexer/src/indexer/indexers/sql.py:310]: And this execution is very fast as compared to for loop. We are inserting all values in a single query.

Comment (romainr) [reply desktop/libs/indexer/src/indexer/indexers/sql.py:310]: Big +1

(Phoenix did not support bulk INSERTs, glad we can do it now with Hive/Impala!)

File: desktop/libs/indexer/src/indexer/indexers/sql_tests.py [hunk 6]
```diff
@@ -603,3 +603,52 @@ def test_create_ddl_with_abfs():
...
+      {'name': 'dest', 'type': 'string'},
+      {'name': 'plane', 'type': 'string'},
+      {'name': 'cancelled', 'type': 'boolean'},
+      {'name': 'time', 'type': 'bigint'},
+      {'name': 'dist', 'type': 'bigint'},
+    ],
+    'indexerPrimaryKey': []
+  }
+  request = MockRequest(fs=MockFs())
+  sql = SQLIndexer(user=request.user, fs=request.fs).create_table_from_local_file(source, destination).get_str()
```
Comment (romainr) [inline desktop/libs/indexer/src/indexer/indexers/sql_tests.py:633]: We should just switch to Mock() at some point

**Review by romainr — APPROVED** [review]

**Pull request merged**